### PR TITLE
ENH: Add VerifyPreconditions and VerifyInputInformation

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

**Comment by blowekamp** [comment]
@N-Dekker Hope this clarifies the behavior.

File: SoftwareGuide/Latex/DevelopmentGuidelines/WriteAFilter.tex [hunk 1]
```diff
@@ -321,12 +321,26 @@ \subsubsection{UpdateOutputInformation()}
...
-terminates at the sources.
+The \code{UpdateOutputInformation()} method first calls the
+\code{VerifyPreconditions} to check that all required inputs
+are set and all parameters are valid and consistent. This enables quick
+failure of a filter when not configure correctly. The default
+implementation checks that all required pipeline inputs are set.
+
+Next the pipeline modified time is determined. The RequestedRegion is
+set to process all the data, i.e., the LargestPossibleRegion, if
+neither \code{UpdateLargestPossibleRegion} was call nor
```
Comment (dzenanz) [inline SoftwareGuide/Latex/DevelopmentGuidelines/WriteAFilter.tex]: was call -> was call**ed**

**Comment by thewtex** [comment]
Rebasing on `master` for CI builds...

**Comment by blowekamp** [comment]
replaced by #121

**Pull request closed without merging**